docs/developer/design/dashboard-architecture.md

```diff
@@ -0,0 +1,660 @@
+    }
+
+    class DataForwarder {
+        +dict[str, DataService] data_services
+        +forward(stream_name, value)
+        +transaction() context
+    }
+
+    class DataSubscriber {
+        <<abstract>>
+        +set[DataKey] keys
+        +trigger(store)
+        +send(data)*
+    }
+
+    class ComponentDataSubscriber {
+        +ComponentDataKey component_key
+        +Pipe pipe
+        +send(data)
+    }
+
+    class DataKey {
+        +str service_name
+        +str source_name
+        +str key
+    }
+
+    class ComponentDataKey {
+        <<abstract>>
+        +str component_name
+        +str view_name
+        +str service_name*
+        +cumulative_key() DataKey
+        +current_key() DataKey
+    }
+
+    DataService --> DataSubscriber : notifies
+    DataForwarder --> DataService : forwards to
+    ComponentDataSubscriber --|> DataSubscriber
+    DataSubscriber --> DataKey : depends on
+    DataService --> DataKey : indexed by
+    ComponentDataSubscriber --> ComponentDataKey : uses
+```
+
+## The BaseParamModel Mechanism
+
+### Purpose and Role
+
+`BaseParamModel` is a key architectural component for simple configuration widgets. It serves as a dedicated translation layer and per-widget controller, bridging the gap between:
+
+- **Param models** (`param.Parameterized`): Used for GUI widgets and user interaction
+- **Pydantic models**: Used for backend validation, serialization, and communication
+
+This mechanism enables a clean, testable, and maintainable way to synchronize widget state with configuration state, without leaking infrastructure details into the presentation layer.
+
+### How It Works
+
+- Each simple configuration widget is backed by a `BaseParamModel` subclass
+- The `BaseParamModel`:
+  - Registers the relevant Pydantic schema with `ConfigService`
+  - Subscribes to config updates for its key, updating the widget state when changes arrive
+  - Propagates user changes from the widget to `ConfigService` by translating Param state to a Pydantic model
+
+```mermaid
+sequenceDiagram
+    participant ConfigService
+    participant BaseParamModel
+    participant PanelWidget
+
+    ConfigService->>BaseParamModel: Notify config update
+    BaseParamModel->>PanelWidget: Update widget state
+    PanelWidget->>BaseParamModel: User changes widget
+    BaseParamModel->>ConfigService: Update config
+```
+
+### Architectural Implications
+
+- **Localized Coupling**: `BaseParamModel` knows about both Param and Pydantic models, but this coupling is intentional and limited to the translation layer
+- **No Architectural Problem**: This is not problematic coupling, but a necessary and well-encapsulated translation between two distinct model types
+- **Testability**: The translation logic is isolated and can be tested independently
+- **Extensibility**: More complex workflows can use a centralized controller, while simple controls benefit from this lightweight mechanism
+
+## MVC Pattern Analysis
+
+### Mapping to MVC
```

Comment:
This relation seems tenuous. Why are services the 'Model' here, but the param models are par tof the controller? 
Does it make sense to compare to MVC?